### PR TITLE
[1LP][RFR] Stacks and Tentants Widgetastic Conversion

### PR DESCRIPTION
**Pull request by rsnyman**
Convert Orchestration Stacks and Tenants to Widgetastic.

Notes:
 - Security groups seems to be unused.
 - Also converted some Python style docstrings to Google style

FIXES RHCFQE-2973

### CODE REVIEW

**Comment by rsnyman** [comment]
Oh, this requires https://github.com/RedHatQE/widgetastic.patternfly/pull/10 to be merged into Widgetastic.patternfly.

**Comment by rsnyman** [comment]
Tests that use Catalogs and Services seem to be failing. I'm not sure why, as I never introduced any changes to that code.

File: cfme/cloud/stack.py [hunk 1]
```diff
@@ -1,34 +1,254 @@
...
+    """The toolbar on the stacks detail page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    lifecycle = Dropdown('Lifecycle')
+    download = Button('Download summary in PDF format')
+
+
+class StackSubpageToolbar(View):
+    """The toolbar on the sub pages, like resources and security groups"""
+    show_summary = Button('Show {} Summary')      # TODO How to get name in there?
```
Comment (quarckster) [inline cfme/cloud/stack.py]: I guess class `Parameter` from `widgetastic.utils` could help here.

File: cfme/cloud/stack.py [hunk 2]
```diff
@@ -41,197 +261,197 @@ def __init__(self, name, provider, quad_name=None, appliance=None):
...
     def edit_tags(self, tag, value):
-        navigate_to(self, 'EditTags')
-        fill(edit_tags_form, {'select_tag': tag,
-                              'select_value': value},
-             action=form_buttons.save)
-        flash.assert_success_message('Tag edits were successfully saved')
+        """Edit the tags of a particular stack
+        :param tag: The tag to set
+        :param value: The value of the tag
+        """
```
Comment (quarckster) [inline cfme/cloud/stack.py]: **REQUIRED**
Inconsistency is here. Below you used `Args` section for arguments and here `:param`.

File: cfme/tests/services/test_provision_stack.py [hunk 3]
```diff
@@ -49,19 +49,33 @@ def template(provider, provisioning, dialog_name, setup_provider):
         data_file = load_data_file(str(orchestration_path.join('azure_vm_template.json')))
 
     template.create(data_file.read().replace('CFMETemplateName', template_name))
+    if provider.type == "azure":
```
Comment (quarckster) [inline cfme/tests/services/test_provision_stack.py]: **REQUIRED**
```python
if provider.one_of(AzureProvider):
```

File: cfme/tests/services/test_provision_stack.py [hunk 4]
```diff
@@ -49,19 +49,33 @@ def template(provider, provisioning, dialog_name, setup_provider):
...
-    if provider.type == "azure":
-        dialog_name = "azure-single-vm-from-user-image"
-    else:
-        dialog_name = "dialog_" + fauxfactory.gen_alphanumeric()
-    yield dialog_name
+def dialog(provider, template):
+    template, dialog_name = template
+    service_name = fauxfactory.gen_alphanumeric()
+    element_data = dict(
+        ele_label="ele_" + fauxfactory.gen_alphanumeric(),
```
Comment (quarckster) [inline cfme/tests/services/test_provision_stack.py]: **REQUIRED**
use `format()` please.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/cloud/stack.py (hunk 1 above)
Comment (rsnyman) [inline cfme/cloud/stack.py]: Just looking at the code for ParametrizedString (the base class for Parameter), it doesn't look like it supports the parameter format I would need. In this particular case I want the "name" attribute of the "object" object within the context.

I usually use this when checking the page title:

```python
expected_title = '{} (Summary)'.format(self.context['object'].name)
```

However, Parameter only seems to support objects within the context object. From widgetastic.utils:

```python
param_value = view.context[context_name]
```

This means that what I would _want_ to do is this:

```python
show_summary = Button(Parameter('Show {object.name} Summary'))
```

But that would result in the following lookup, which wouldn't work:

```python
param_value = view.context['object.name']
```

Comment (mfalesni) [inline cfme/cloud/stack.py]: @rsnyman We can work on this, this is doable :) I can extend the formatter to detect `.` and then do a nested lookup. Do you have any other ideas?

Comment (rsnyman) [inline cfme/cloud/stack.py]: @mfalesni That would be awesome, I've wanted to use this in a few scenarios. I don't have any other ideas right now.

**Comment by rsnyman** [comment]
`upstream` fails because Sprout is unable to provision an appliance.

**Comment by psav** [comment]
Great work!

**Pull request merged**